src/c++/library/http_client.cc

```diff
@@ -538,7 +539,7 @@ HttpInferRequest::GetNextInput(uint8_t* buf, size_t size, size_t* input_bytes)
   }
 
   while (!data_buffers_.empty() && size > 0) {
-    const size_t csz = (std::min)(data_buffers_.front().second, size);
+    const size_t csz = std::min(data_buffers_.front().second, size);
```

Comment:
was this causing a problem with Windows?

Reply:
Yes. This is causing the compilation syntax error on windows. `(std::min)()` was a previous fix to get around the issue, but there are two additional calls below it that does not wrap `std::min` in `()`. To make the fix a bit more readable, we can unify the fix to use `#undef min` for windows on top, and "undo" the existing fix.